spec/function.dd

```diff
@@ -2524,6 +2524,288 @@ $(H3 $(LNAME2 this-reference, `this` Reference))
         )
 
 
+
+$(H2 $(LNAME2 refscopereturn, Ref Scope Return Cases))
+
+
+$(H3 $(LNAME2 rsr_definitions, Definitions))
+
+    $(DL
+    $(DI I) $(DD type that contains no indirections)
+    $(DI P) $(DD type that contains indirections)
+    $(DI X) $(DD type that may or may not contain indirections)
+    $(DI p) $(DD parameter of type P)
+    $(DI i) $(DD parameter of type I)
+    $(DI ref) $(DD `ref` or `out` parameter)
+
+    $(DI returned) $(DD returned via the `return` statement)
+    $(DI escaped) $(DD stored in a global or other memory not in the function$(RSQUO)s stack frame)
+    )
+
+$(H3 $(LNAME2 rsr_classification, Classification))
+
+$(P A parameter must be in one of the following states:)
+
+    $(TABLE2 Classification,
+    $(THEAD Term, Description)
+    $(TROW None,
+        `p` may be returned or escaped)
+
+    $(TROW ReturnScope,
+        `p` may be returned but not escaped)
+
+    $(TROW Scope,
+        `p` may be neither returned nor escaped)
+
+    $(TROW Ref,
+        `p` may be returned or escaped$(LF)
+        `ref` may not be returned nor escaped)
+
+    $(TROW ReturnRef,
+        `p` may be returned or escaped$(LF)
+        `ref` may be returned but not escaped)
+
+    $(TROW RefScope,
+        `p` may be neither returned nor escaped$(LF)
+        `ref` may not be returned nor escaped)
+
+    $(TROW ReturnRef-Scope,
+        `p` may be neither returned nor escaped$(LF)
+        `ref` may be returned but not escaped)
+
+    $(TROW Ref-ReturnScope,
+        `p` may be returned but not escaped$(LF)
+        `ref` may not be returned nor escaped)
+
+    $(TROW ReturnRef-ReturnScope,
+        `p` may be returned but not escaped$(LF)
+        `ref` may be returned but not escaped$(LF)
+        This isn't expressible with the current syntax
+        and so is not allowed.)
+    )
+
+$(H3 $(LNAME2 rsr_mapping, Mapping Syntax Onto Classification))
+
+    $(P The juxtaposition of `return` immediately preceding `scope` means ReturnScope.
```

Comment:
This is already covered by another paragraph on this page

Reply:
deleted the other

Reply:
Here's the first appearance:
https://github.com/dlang/dlang.org/blob/e60a7f08b67377d289cd871090b140ae11459fcf/spec/function.dd#L1844-L1846